homeassistant/components/manual/alarm_control_panel.py

```diff
@@ -385,9 +386,9 @@ def device_state_attributes(self):
         """Return the state attributes."""
         state_attr = {}
 
-        if self.state == STATE_ALARM_PENDING:
-            state_attr[ATTR_PRE_PENDING_STATE] = self._previous_state
```

Comment:
Can't these attributes be used already to accomplish the motivation of the PR?  To distinguish between pending arming and pending triggering?

Reply:
Yes they can, moreover since this is a manual alarm you probably also already have an automation to trigger the alarm in the first place (I do) so you could also add an action to that automation to trigger the TTS warning as discribed in the PR example.

There are a ton of ways to acomplish the example I presented, but I think the only proper nice way that is clear for all users would be changing the state to "arming" and 'pending" which is in my opinion just more clear.

However I do not need this PR to get my setup working, just want to improve it to make it more clear for people in general and for the future.

Reply:
I agree, the current state is confusing IMO.  I just wanted to think through exactly why this PR should be implemented.  It isn't about adding functionality, it is to make the existing behavior clearer.

Reply:
Could these attributes be simply removed now if pending is split up?  I suppose you could have automations that depend on which state it is arming to our triggering from.  But I'm not sure we need the pre-arming state, is it always `disarmed`? And the post-pending state, is it always `triggered`?

Reply:
The ATTR_PREVIOUS_STATE is actually used to restore the previous state on startup of HomeAssistant if homeassistant closed during a transition of a state ("arming" or "pending"), it is at the bottom of the code.

The ATTR_NEXT_STATE is not used inside the code, so that is purely for people wanting to make automations that are diffrent when for instance "arming" to "armed_away" or "armed_home". I don't use that so I would be happy to remove ATTR_NEXT_STATE , but I think that might upset some people...